src/wavm.cpp

```diff
@@ -0,0 +1,263 @@
+#include "wavm.h"
+#include "debugging.h"
+#include "eei.h"
+#include "exceptions.h"
+
+#pragma GCC diagnostic ignored "-Wunused-parameter"
+
+using namespace std;
+
+namespace hera {
+
+
+std::unique_ptr<WasmEngine> WavmEngine::create()
+{
+  return std::unique_ptr<WasmEngine>{new WavmEngine};
+}
+
+
+namespace wavm_host_module {
+
+  // first a pointer to the ethereum interface, which is used in host functions
+  WavmEthereumInterface* interface;
```

Comment:
This cannot stay like this. The `WavmEthereumInterface` doesn't only hold helpers for the host functions, but the call context they are in.

Having this as a global means any ewasm bytecode doing a call to another contract wouldn't work.

Reply:
I see. Wavm and Hera architectures seem incompatible.

Reply:
Just open an issue upstream asking for a context to be passed down.

Reply:
They will point me to the existing way to pass something to host functions, but it is complicated with mutexes, garbage collectors, custom create functions instead of constructors. I will try this but might take time and might be unsuccessful.